common/persistence/sql/sqlplugin/history_timer_tasks.go

```diff
@@ -0,0 +1,71 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sqlplugin
+
+import (
+	"database/sql"
+	"time"
+)
+
+type (
+	// TimerTasksRow represents a row in timer_tasks table
+	TimerTasksRow struct {
+		ShardID             int32
+		VisibilityTimestamp time.Time
+		TaskID              int64
+		Data                []byte
+		DataEncoding        string
+	}
+
+	// TimerTasksFilter contains the column names within timer_tasks table that
+	// can be used to filter results through a WHERE clause
+	TimerTasksFilter struct {
```

Comment:
Are all three fields always required? Same question as with `TransferTasksFilter` do we even need this struct anymore or should we pass the three arguments directly to the methods instead?

Reply:
all 3 are required for pagination

Reply:
and get timer (one and only one)

Reply:
If others are for this convention of using a struct for passing arguments into internal methods, I have no objection.

Reply:
create an issue #810 so we can revisit the naming convention before make the persistence layer modular for override